plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotMetadata.java

```diff
@@ -294,6 +306,211 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
         return Optional.of(new ConstraintApplicationResult<>(handle, remainingFilter, false));
     }
 
+    @Override
+    public Optional<AggregationApplicationResult<ConnectorTableHandle>> applyAggregation(
+            ConnectorSession session,
+            ConnectorTableHandle handle,
+            List<AggregateFunction> aggregates,
+            Map<String, ColumnHandle> assignments,
+            List<List<ColumnHandle>> groupingSets)
+    {
+        if (!isAggregationPushdownEnabled(session)) {
+            return Optional.empty();
+        }
+
+        // Global aggregation is represented by [[]]
+        verify(!groupingSets.isEmpty(), "No grouping sets provided");
+
+        if (groupingSets.size() != 1) {
+            return Optional.empty();
+        }
+
+        PinotTableHandle tableHandle = (PinotTableHandle) handle;
+        Optional<AggregationApplicationResult<ConnectorTableHandle>> distinctCountHllResult = applyDistinctCountHll(session, tableHandle, aggregates, assignments, groupingSets);
+        if (distinctCountHllResult.isPresent()) {
+            return distinctCountHllResult;
+        }
+        if (tableHandle.getQuery().isPresent() &&
+                !(tableHandle.getQuery().get().getAggregateColumns().isEmpty() &&
+                        tableHandle.getQuery().get().getGroupingColumns().isEmpty())) {
+            return Optional.empty();
+        }
+
+        ImmutableList.Builder<ConnectorExpression> projections = ImmutableList.builder();
+        ImmutableList.Builder<Assignment> resultAssignments = ImmutableList.builder();
+        ImmutableList.Builder<AggregationExpression> aggregationExpressions = ImmutableList.builder();
+
+        for (AggregateFunction aggregate : aggregates) {
+            Optional<String> inputExpression = getInputExpression(aggregate, assignments);
+            // Do not push COUNT(<column>) into pinot as it converts to COUNT(*)
+            // This will count null values.
+            if (inputExpression.isEmpty() || (isCountFunction(aggregate) && !isCountStarFunction(aggregate))) {
+                return Optional.empty();
+            }
+            // Distinct aggregations other than distinctcounthll are not supported
+            if (aggregate.isDistinct()) {
+                return Optional.empty();
+            }
```

Comment:
This should be handled for individual functions separately. What would it take to use `AggregateFunctionRewriter` here?

Also, checking `isDistinct` is not enough.
See `io.trino.plugin.jdbc.expression.AggregateFunctionPatterns#basicAggregation` which matches only "basic" aggregation functions, without features that are both:  easy to forget about and hard to implement in the remote system.